tillerinobot/src/main/java/tillerino/tillerinobot/CommandHandler.java

```diff
@@ -248,4 +250,50 @@ public String getChoices() {
 			}
 		};
 	}
+
+	public static abstract class WithCommand implements CommandHandler {
+		private final String command;
+		private final String alias;
+		private final String aliasWithSpace;
+
+		public WithCommand(String command) {
+			this.command = command;
+			this.alias = String.valueOf(command.charAt(0));
+			this.aliasWithSpace = new String(new char[]{command.charAt(0), ' '});
+		}
+
+		@Override
+		public final Response handle(String originalCommand, OsuApiUser apiUser, UserData userData) throws UserException,
+				IOException, SQLException, InterruptedException {
+			String lowerCase = originalCommand.toLowerCase();
+			final String remaining;
+			searchCommand: {
+				if (lowerCase.equals(alias)) {
+					remaining = "";
+					break searchCommand;
+				}
+				if (getLevenshteinDistance(lowerCase, command) <= 2) {
+					remaining = "";
+					break searchCommand;
+				}
+				if (lowerCase.startsWith(aliasWithSpace)) {
+					remaining = originalCommand.substring(2);
+					break searchCommand;
+				}
+				if (lowerCase.contains(" ")) {
+					int pos = lowerCase.indexOf(' ');
+					if (getLevenshteinDistance(lowerCase.substring(0, pos), command) <= 2) {
+						remaining = originalCommand.substring(pos + 1);
+						break searchCommand;
+					}
+				}
+				return null;
+			}
+
+			return handleCommand(remaining, apiUser, userData);
+		}
+
+		public abstract Response handleCommand(String remaining, OsuApiUser apiUser, UserData userData) throws UserException,
```

Comment:
I think this should be renamed to `handleArgument` to make it clear what's happening.